src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/Engine/TokenPairWithOperations.cs

```diff
@@ -13,16 +13,16 @@ namespace Microsoft.CodeAnalysis.Formatting
     internal readonly struct TokenPairWithOperations
     {
         public readonly TokenStream TokenStream;
-        public readonly AdjustSpacesOperation? SpaceOperation;
-        public readonly AdjustNewLinesOperation? LineOperation;
+        public readonly AdjustSpacesOperation SpaceOperation;
+        public readonly AdjustNewLinesOperation LineOperation;
```

Comment:
📝 This was the meaningful change. A very large collection of `TokenPairWithOperations` gets created, and the use of `Nullable<T>` was expanding each of these fields from 8 bytes to 12 bytes on x64, which increased the size of the collection by ~600MB. Converting these two fields to use `AdjustSpacesOperation.None` and `AdjustNewLinesOperation.None` instead of null was a small and easy changes. All of the other related changes were simply an effort to avoid having two different representations of exactly the same concept in code.